### PR TITLE
Change inline code styling.

### PR DESCRIPTION
**Pull request by sfc-gh-tteixeira**
## 🧠 Description of Changes

**Revised:**
Screenshot 2024-12-18 at 09 41 24

**Current:**
Screenshot 2024-12-18 at 09 41 38

## 💥 Impact

Size:

- [ ] Small 
- [ ] Not small 

## 🌐 References

- [ ] [Notion](...)

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

### CODE REVIEW

**Comment by sfc-gh-tteixeira** [comment]
Example pages to check:
* /develop/api-reference/data/st.dataframe
* /get-started/installation/command-line
* /develop/api-reference/media/st.logo (see link)

File: styles/text.scss [hunk 1]
```diff
@@ -109,8 +109,23 @@ button:focus-visible {
...
+}
+
+/* Improve `code` links. Don't use border, use underline instead. */
+p > a:has(tt.docutils.literal) {
+  @apply border-0;
+}
+
+/* Improve `code` links. Don't use border, use underline instead. */
+p > a > tt.docutils.literal {
+  @apply underline;
```
Comment (sfc-gh-dmatthews) [inline styles/text.scss:128]: If we use underline, commands with underscores will look like this:
image

Also, the site uses border for links elsewhere, so if we do switch it, we have other styling to correct too. This is what the preview currently shows:

A link in the function description of st.logo:
image

A link in the class description of SQLConnection:
image

Links from the connections concept page (both standard and header-code):
image


The only thing that's been changed is inline code within a parameter description, I think.

Comment (sfc-gh-dmatthews) [reply styles/text.scss:128]: I'd vote for leaving it as a border to be consistent across all links and just modify the color if that's what you want. Though I'd be a little confused by the color scheme if we make it a green underline for code (to always match the text color), then how do we handle the hover effect. We currently hover black -> red for the bottom border on links, but green -> red feels....seasonal. 😅

Comment (sfc-gh-tteixeira) [reply styles/text.scss]: For regular links we'll continue to use the existing design. What we're changing here is the design of links inside inline code from autofunctions.

So from your examples, only the SQL Connection example will be impacted by this PR. The rest will stay the same.

Comment (sfc-gh-dmatthews) [reply styles/text.scss]: In the connection to data article, `st.connection` is also inline code; it's just not inline code within Autofunction. Do you mean to exclude that as well?

**Comment by sfc-gh-dmatthews** [comment]
Tips, Info, Warning: e.g. https://deploy-preview-1203--streamlit-docs.netlify.app/develop/concepts/multipage-apps/overview
image


So we have inline code is green except:
* Inline code in headers is black
* Inline code with links in MarkDown (not AutoFunction) is black with a bottom border (matches non-inline-code links)
* Inline code with links in AutoFunciton are green underlined

Is that the correct intent?

**Comment by sfc-gh-dmatthews** [comment]
Similarly for deprecation/note blocks within Autofunction, can you confirm if this is how you want the background colors to look?
image
image

File: styles/text.scss [hunk 2]
```diff
@@ -109,8 +109,25 @@ button:focus-visible {
...
+/* Improve `code` links. Don't use border, use underline instead. */
+a:has(tt:only-child),
+a:has(code:only-child) {
+  @apply border-0;
+}
+
+/* Improve `code` links. Don't use border, use underline instead. */
+a > tt:only-child,
+a > code:only-child {
+  @apply underline;
```
Comment (sfc-gh-dmatthews) [inline styles/text.scss:130]: Inline code links outside of Autofunction have double underline in default text color:
image

Inline code links in Autofunction (function or class description; not parameter description) have a double underline (underline is green, border is default text color)
image

Comment (sfc-gh-tteixeira) [reply styles/text.scss:130]: Great finds! Should be fixed now.

**Comment by sfc-gh-dmatthews** [comment]
This link contains inline code but isn't entirely inline code. Do we just declare that I can't do that anymore since we're using different underlines for code and non-code?

image

**Comment by sfc-gh-tteixeira** [comment]
No, that should be allowed. The problem here seems to be that `:only-child` doesn't count text nodes as children -- weird! So we'll have to look for another solution...

**Comment by sfc-gh-tteixeira** [comment]
Ok, I tried different solutions but nothing look good. Reverting to the way links used to look.

I don't love it, but it's the best we can do in short notice.

Screenshot 2024-12-17 at 15 53 43

**Review by sfc-gh-dmatthews — APPROVED** [review]

**Pull request merged**